src/distributions/range.rs

```diff
@@ -88,6 +88,7 @@ impl<X: SampleRange> Distribution<X> for Range<X> {
 /// Helper trait for creating objects using the correct implementation of
 /// `RangeImpl` for the sampling type; this enables `Range::new(a, b)` to work.
 pub trait SampleRange: PartialOrd+Sized {
+    /// Actual `RangeImpl` implementation for `X`.
```

Comment:
I'd prefer you didn't use 'actual'; try "The `RangeImpl` implementation supporting type `X`."

Also, it would make sense now to rename `X` → `T` and `T` to something else (perhaps `Impl`; it's not something users will see a lot). This is what Huon suggested I think.

Reply:
That sounds better, as always.